### PR TITLE
Feat set seed

### PR DESCRIPTION
**Pull request by LeonardoViotti**
Feature to set random state for reproducible training. Solving #576

### CODE REVIEW

File: opensoundscape/seed.py
Comment (sammlapp) [inline opensoundscape/seed.py:1]: I think this can go in opensoundscape/utils.py rather than being a separate module

File: tests/test_seed.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+from opensoundscape import seed
+import numpy as np
+import torch
+import random
+from opensoundscape.ml import cnn, cnn_architectures
+
+import pytest
+
+pytestmark = pytest.mark.parametrize("input", [1, 11, 13, 42, 59, 666, 1234])
```
Comment (sammlapp) [inline tests/test_seed.py:9]: what does this do?

File: opensoundscape/seed.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+""" seed.py: Set random state across different libraries for reproducibility
+"""
+import numpy as np
+import torch
+import random
+
+
+def seed(seed, verbose=True):
```
Comment (sammlapp) [inline opensoundscape/seed.py:8]: needs a docstring, with Args and Return fields; see other functions for examples

File: opensoundscape/seed.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    print(f"Random state set with seed {seed}")
```
Comment (sammlapp) [inline opensoundscape/seed.py:9]: did you mean to wrap this in `if verbose:` ? `verbose` is currently unused

**Review by sammlapp — CHANGES_REQUESTED** [review]

File: opensoundscape/seed.py
Comment (LeonardoViotti) [reply opensoundscape/seed.py:1]: Done, but Let the unit test separate because I didn't want to mess-up test_utils.py.

File: tests/test_seed.py (hunk 1 above)
Comment (LeonardoViotti) [reply tests/test_seed.py]: This is a global version of the @pytest.mark.parametrize decorator to use the list as inputs for all functions in the script. See https://docs.pytest.org/en/7.3.x/how-to/parametrize.html for more explanation.

File: opensoundscape/seed.py (hunk 2 above)
Comment (LeonardoViotti) [reply opensoundscape/seed.py]: Done

File: opensoundscape/seed.py (hunk 3 above)
Comment (LeonardoViotti) [reply opensoundscape/seed.py]: Yes, I did. Just changed that.

File: opensoundscape/utils.py [hunk 4]
```diff
@@ -329,3 +330,25 @@ def generate_opacity_colormaps(
         colormaps.append(cmap)
 
     return colormaps
+
+
+import numpy as np
```
Comment (sammlapp) [inline opensoundscape/utils.py:335]: imports should always be at the top of a file. VS Code will help detect duplicate imports

File: tests/test_utils.py [hunk 5]
```diff
@@ -168,3 +171,66 @@ def test_make_clip_df_from_label_df(silence_10s_mp3_str, metadata_wav_str):
...
+
+    utils.set_seed(input)
+    model_resnet2 = cnn_architectures.resnet18(num_classes=10, weights=None)
+    lw2 = model_resnet2.layer1[0].conv1.weight
+
+    utils.set_seed(input + 1)
+    model_resnet3 = cnn_architectures.resnet18(num_classes=10, weights=None)
+    lw3 = model_resnet3.layer1[0].conv1.weight
+
+    assert torch.all(lw1 == lw2) & torch.any(lw1 != lw3)
```
Comment (sammlapp) [inline tests/test_utils.py:236]: don't understand the logic here, can you add a comment? is the `any` asserting that _none_ of the weights should be the same? That doesn't seem like the correct logic, rather we want to assert that the entire array isn't equal to the other?

File: opensoundscape/utils.py (hunk 4 above)
Comment (LeonardoViotti) [reply opensoundscape/utils.py]: Oh, sorry, missed that. Just pushed the correction.

File: tests/test_utils.py (hunk 5 above)
Comment (LeonardoViotti) [reply tests/test_utils.py:239]: In summary, also assert that at least one element is different if different seeds. Added as a comment in [tests/test_utils.py](https://github.com/kitzeslab/opensoundscape/blob/1fc6d3a00197f9141ec5d92efdbf4d5bb4fe81bd/tests/test_utils.py#L181).
